3.8 Documentation refresh

SimonDarksideJ opened the pull request:
First pass at a documentation refresh for the MonoGame 3.8 Release

* Stylised the previous 3.7 project setup guides (retained for now due to the breaking change update with 3.8 tooling)
* Resolves: #7237 - pipeline tool registration command fixed
* Added new 3.8 project setup guides for Visual Studio, Visual Studio for Mac and .NET CLI
* Updated the other getting starting guides for 3.8
* Other minor docs updates

Still to do:
[X] Grammar and content review
[X] Run through each tutorial on a clean machine for accuracy
[X] Address any review feedback
[X] Update MGCB-Editor Images

tomspilman commented:
Here is a link to these rebuilt docs:

http://teamcity.monogame.net/repository/download/MonoGame_GenerateDocumentation/99430:id/Documentation.zip!/index.html

tomspilman commented:
> Stylised the previous 3.7 project setup guides (retained for now due to the breaking change update with 3.8 tooling)

@Jjagg  @harry-cpp - Do we really want to keep 3.7 setup guides in the 3.8 docs?  Seems to me we should ditch that and keep the old 3.7 docs in a separate place.

harry-cpp commented:
> @Jjagg @harry-cpp - Do we really want to keep 3.7 setup guides in the 3.8 docs? Seems to me we should ditch that and keep the old 3.7 docs in a separate place.

Agreed.

tomspilman commented:
Like maybe `https://docs.monogame.net/3.7/` vs `https://docs.monogame.net/` ?

SimonDarksideJ commented:
I'd say as 3.8 is a breaking change from the way that MonoGame is installed and people will still need 3.7 to maintain their old project, we should keep it for now.
I agree, if we have something like https://docs.monogame.net/3.7/ and https://docs.monogame.net/3.8/ with https://docs.monogame.net pointing to the current version would be cleaner

tomspilman commented:
> I agree, if we have something like

@Jjagg - how bad would it be setting up different locations for doc versions?  We can at least do this for 3.7, 3.8 and new docs moving forward.  If we really wanted we could pull old docs from history and rebuild them to get back 3.6 or older docs.

SimonDarksideJ commented:
If you let me know what to put where in this PR I'll get it updated, unless there is something on the build side that needs updating?

roy-t commented:
I just noticed that in the documentation (http://teamcity.monogame.net/repository/download/MonoGame_GenerateDocumentation/99430:id/Documentation.zip!/articles/getting_started/3.8/1_creating_a_new_project_vs.html)  it says "dotnet tool install -g dotnet-mgcb" but then the quote about development branches it says  "dotnet tool install -g dotnet-mgcb-editor --version 3.8.0.1375-develop" notice the dotnet-mgcb vs dotnet-mgcb-editor. From what I've tried it seems that the version without -editor works fine.

harry-cpp commented:
Refer to Pipeline Tool as MGCB Editor so there is no confusion. 

> From what I've tried it seems that the version without -editor works fine.

`dotnet-mgcb` only installs the content compiler, not the UI tool.

SimonDarksideJ commented:
OK, I'll go back and make those changes to the docs while the folder discussion continues

SimonDarksideJ commented:
Resolved identified issues
* Pipeline Editor should be names and referenced only as the MGCB editor
* dotnet commands for installing the MGCB editor have been fixed

Although with the MGCB naming update in the documentation, can we rename the EXE / and Window to reflect this, else we have inconsistent naming of the tools from the documentation for those tools. 
Thoughts @tomspilman / @harry-cpp ?

Jjagg commented:
> how bad would it be setting up different locations for doc versions? 

This is really easy, we just need to extract the docs in a subfolder with the path we want them at. I'm pretty sure all local links in the docs are relative so it should just work.

SimonDarksideJ commented:
OK, I'll reorganise the folders to match the above spec

tomspilman commented:
> OK, I'll reorganise the folders to match the above spec

What does that mean?  I don't think you need to change folders in this PR at all.

The fix to support multiple docs per version is in how we deploy the docs to the website.

SimonDarksideJ commented:
OK, I'll keep it as is for now.  Just need to refresh the MGCB-Editor images in the docs and this will be complete

SimonDarksideJ commented:
One thought @Jjagg 

> I'm pretty sure all local links in the docs are relative so it should just work.

Images all come from the root of the docs folder. Which should still be fine, except it would need each image to be version dependant maybe.  Thoughts?

SimonDarksideJ commented:
* Refreshed all pipeline images to MGCB
* Major update to MGCB-Editor tutorial
* Moved images for the "Content" section to a new content folder.

Will wait for the build and do another review pass

willmotil commented:
@SimonDarksideJ Simon yesterday i submitted a pull request to make a minor amendment to the instructions on the documentation for the tool page which has a minor omission but it seems to have disappeared? Is this because this is all being moved around? Should i wait resubmit it ?  Or is it still here somewere im not seeing it when i searched for it.

SimonDarksideJ commented:
Not seeing your edit suggestions @willmotil , either post the changes here (it is a small file) or do a new PR against the original branch for this change here:
https://github.com/SimonDarksideJ/MonoGame/tree/dev/docsupdate

SimonDarksideJ commented:
PR ready to review and merge

SimonDarksideJ commented:
Link to docs to review:
http://teamcity.monogame.net/repository/download/MonoGame_GenerateDocumentation/99758:id/Documentation.zip!/index.html

* All links checked
* All images as expected
* Grammar and proofing done

tomspilman commented:
@SimonDarksideJ - One last thing...

![image](https://user-images.githubusercontent.com/103522/88539966-ce0b8080-cfd7-11ea-9e69-eac259e9fc12.png)

Can you update this introduction to point people at the "Articles" and "APIs" sections?  I am concerned the landing page not showing the articles TOC on the left will lead people to not know where to start.

SimonDarksideJ commented:
THat's a fair ask @tomspilman , originally I was only looking to refresh the Getting Started guides, but I'll do a quick wider review to see what else we can update.

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
 
 This area of the site contains the documentation on the API of MonoGame as well as how to use it to create great games.
-Note that this is a work in progress so there will be gaps in the documentation coverage. 
+Note that this is a work in progress so there will be gaps in the documentation coverage.
```
Maybe the second sentence of the docs shouldn't be "hey the docs suck". ;)

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
+not be available in the current stable release.
 
-### We Need Your Help!
+---
```
Lets remove this... it is ugly.

SimonDarksideJ replied:
:D

SimonDarksideJ replied:
The entire section or just the header @tomspilman ?

tomspilman replied:
Just meant the `---` ... they don't make anything more readable.  Use newlines if we need to space things out more.

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
+with [XNA](https://msdn.microsoft.com/en-us/library/bb203940.aspx) even down to the namespaces. So usually what works for XNA will work for MonoGame too.
 
-Note that this documentation hub is built from the source code on every commit to the development branch. As such 
+> Note that this documentation hub is built from the source code on every commit to the development branch. As such
```
Lets remove this notice and fix the development docs to be separate from the release docs.

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
+---
 
-Truly great open source projects require high quality documentation.  This is call for volunteers to help us make the MonoGame documentation truly great.  If you can help write tutorials, guides, code snippets, reference docs, video walkthroughs or just any improvement to our current documentation we could use your help!
+## Site Index
```
Lets remove this title.

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
+
+The documentation site is made up of several major parts which are available as links at the top of the page.  If you are new to MonoGame, we highly recommend walking through the **[Getting Started](articles/getting_started/getting_started)** section to Install MonoGame and build your first game.
+
+### [Articles](/articles/introduction/introduction.md)
```
Make the `Articles` and `API Reference` bullet points with descriptions like Microsoft docs do:

![image](https://user-images.githubusercontent.com/103522/88585558-d7690d00-d018-11ea-82b5-e4cfeff65511.png)

SimonDarksideJ replied:
Great idea @tomspilman 
Will also pull the text from the original site for prosperity (as shown above)

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
+
+In the articles section, you will find resources and guides for how to make the most out of the MonoGame Framework, this includes:
+
+* [Introductions](articles/introduction/introduction.md)
```
Sorry i wasn't clearer...  you seem to have misinterpreted what i said.  I don't want a manual TOC to maintain on the main page.  Just a little info about the different major sections of the docs.

So remove most all of this  TOC section stuff.

At most we could maybe keep the major top level sections.... Introduction, Getting Started, Adding Content, Tools, etc.

SimonDarksideJ replied:
Ok fair enough.  I only kept the top-level sections as they are the most likely targets for new users.  But agree, let's keep it simple

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
+
+* [MonoGame.Framework.Utilities](https://docs.monogame.net/api/MonoGame.Framework.Utilities.html)
+
+---
```
We really don't need these HRs... they look ugly and don't make things more readable.

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,88 @@
+
+---
+
+### [Changelog](https://docs.monogame.net/CHANGELOG.html)
```
Like the others above... this should be a bullet point with a description and not a header.

SimonDarksideJ commented:
Feedback addressed @tomspilman , hope this is more to your liking.  Agree, we should replicate the original XNA site where possible.

SimonDarksideJ commented:
Updated Index page
![image](https://user-images.githubusercontent.com/1793042/88589156-9ab90800-d050-11ea-9fd7-935855535d36.png)

Jjagg commented:
By the way, with docfx we can put [inline notes](https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#note-warningtipimportant) in our docs. Not sure where that’s applicable, but I’m sure they would fit nicely for emphasis in some places. They might need some custom css to look good with the rest of our styling. Just putting it out there :)

tomspilman commented:
> By the way, with docfx we can put inline notes in our docs

Cool... so a little enhanced markdown we can use.  That is nice and we should use it where appropriate for critical stuff.

SimonDarksideJ commented:
Will keep that in mind for any future tutorials we can add. Especially from my book writing days, having custom highlights really makes things pop.

I'd like to revisit the "Getting Started" area and take more of a feed from the original XNA "Getting Started" tutorials and translate them for MonoGame.  But "later".

For now, this should be ready for 3.8 unless there are any further comments

SimonDarksideJ commented:
Don't know if you spotted it @tomspilman but I also updated the Help and Support page to include the Gitter and Discord channels
http://teamcity.monogame.net/repository/download/MonoGame_GenerateDocumentation/99876:id/Documentation.zip!/articles/introduction/help_and_support.html

![image](https://user-images.githubusercontent.com/1793042/88596316-bde9b480-d05c-11ea-8db6-e2a81c86bccd.png)

tomspilman commented:
>  I also updated the Help and Support page to include the Gitter and Discord channels

Cool...  but what is this artifact?

![image](https://user-images.githubusercontent.com/103522/88597203-5b78c000-d02c-11ea-85f1-9f1d10043c60.png)

Is it like a table to align things like that?  Maybe just stack them vertically instead to avoid it.

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,46 @@
+  
+  Provides helpful information and requirements for developing games with MonoGame.
+
+* [Getting Started guides](articles/getting_started/getting_started.md)
```
Drop the word `guides`

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,46 @@
+
+  Provides introductory information about installing and getting started with developing games using MonoGame.
+
+* [Platform Specific help](articles/platform_specific/platform_specific.md)
```
Let's change it to just `Platform Notes`.

tomspilman replied:
Also related to this....

![image](https://user-images.githubusercontent.com/103522/88600989-2d4bae00-d035-11ea-8d40-15dc09e839d1.png)

... be good for the case be consistent across all docs (using normal headline casing rules).  Also don't think a hyphen is need there.

SimonDarksideJ replied:
Sorted

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,46 @@
+
+  The MonoGame Framework class library is a library of classes, interfaces, and value types that are included in MonoGame.
+
+* [Changelog](https://docs.monogame.net/CHANGELOG.html)
```
Make this `MonoGame Release History`

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,46 @@
+
+* [Changelog](https://docs.monogame.net/CHANGELOG.html)
+
+  For full details on the [MonoGame release history](https://docs.monogame.net/CHANGELOG.html), all the way back to the beginnings of MonoGame.
```
No need for the duplicate link here... you can remove it.  

`For full details on changes all the way back to the beginnings of MonoGame.`

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,46 @@
+
+  For full details on the [MonoGame release history](https://docs.monogame.net/CHANGELOG.html), all the way back to the beginnings of MonoGame.
+
+> Don't forget to check out the [official MonoGame full game samples](articles/samples.md) and the recognized [contributor blogs and tutorials](articles/tutorials.md).
```
Make these two bullet points to match the formatting above.  Samples and Demos` and `Community Tutorials`.

tomspilman commented on Documentation/index.md:
```diff
@@ -1,17 +1,46 @@
+The documentation site is made up of several major parts which are available as links at the top of the page.  
 
-### We Need Your Help!
+> If you are new to MonoGame, we highly recommend walking through the **[Getting Started](articles/getting_started/getting_started.md)** section to Install MonoGame and build your first game.
```
Not sure we need this extra block quote here... we have the getting started guide linked just a line or two below.

tomspilman commented on Documentation/articles/getting_started/getting_started.md:
```diff
@@ -1,10 +1,30 @@
+  - [Visual Studio for Mac](3.8/1_creating_a_new_project_vsm.md)
+  - [DotNet core CLI](3.8/1_creating_a_new_project_netcore.md)
+
+- 3.7
```
We should remove 3.7 docs... we will publish the old 3.7 docs separately later.

SimonDarksideJ replied:
Fair enough

tomspilman commented on Documentation/articles/getting_started/getting_started.md:
```diff
@@ -1,10 +1,30 @@
+
+## Creating a new Project
+
+- 3.8
```
These docs are all for 3.8... lets remove references to 3.8 and move the docs out of a 3.8 folders in the links below.

SimonDarksideJ replied:
3.7 and all references to it be gone.

tomspilman commented on Documentation/articles/getting_started/getting_started.md:
```diff
@@ -1,10 +1,30 @@
+
+## [Migrating from MonoGame 3.7 to 3.8?](/migrate38.md)
+
+If you currently have a MonoGame 3.7 project and wish to upgrade it to MonoGame 3.8, please check our handy [Upgrade Guide](/migrate38.md).
```
It's ok to have a migration from 3.7 guide for the next few versions of MG since we're making the big move from installers to NuGet.  That said... these docs don't seem to exist... the link doesn't work.

Jjagg replied:
Needs to be `~/articles/migrate38.md`: https://docs.monogame.net/articles/migrate38.html

SimonDarksideJ replied:
Fixed

tomspilman commented on Documentation/articles/getting_started/getting_started.md:
```diff
@@ -1,10 +1,30 @@
 - [Adding Content](3_adding_content.md)
 - [Adding Basic Code](4_adding_basic_code.md)
+
+## [Migrating from MonoGame 3.7 to 3.8?](/migrate38.md)
```
Avoid making headers into links... it is sort of ugly and unnecessary if we plan to link it again in the description below.

tomspilman commented on Documentation/articles/getting_started/toc.yml:
```diff
@@ -1,10 +1,19 @@
+    href: 3.8/1_creating_a_new_project_vsm.md
+  - name: MonoGame netcore CLI
+    href: 3.8/1_creating_a_new_project_netcore.md    
+- name: Creating a New Project with MonoGame 3.7
```
Same... 3.7 docs should go.

tomspilman commented:
Thanks @SimonDarksideJ ....  things are looking much nicer!  Found some more little things and putting up further notes.

tomspilman commented:
@Jjagg you happen to know why this doc has a `In this article`....

![image](https://user-images.githubusercontent.com/103522/88600850-dcd45080-d034-11ea-9a82-fb546c1e93d6.png)

... but none of the ones in the articles list have it...

![image](https://user-images.githubusercontent.com/103522/88600915-fa091f00-d034-11ea-84e8-17ca0a9abe71.png)

tomspilman commented on Documentation/articles/introduction/help_and_support.md:
```diff
@@ -8,9 +8,8 @@ If you want to find an answer to a more specific problem, you can ask it on our
-| | |
-|--|--|
 |[![Gitter](~/images/third_party/gitterlogo.jpg)](https://gitter.im/MonoGame/MonoGame) |[![Discord](~/images/third_party/discordlogo.png)](https://discord.gg/xCehJFD) |
+|--|--|
```
Hah... now the issue moved below the images...

![image](https://user-images.githubusercontent.com/103522/88601360-1bb6d600-d036-11ea-9e63-1958fc644fb0.png)

... i think we just need to stack them and not worry about centering them.

Jjagg replied:
Can also use HTML if we want to.

SimonDarksideJ replied:
OK, will stack them :D  Was hoping that altering the markdown would be enough, but I guess not :D

Jjagg commented:
> @Jjagg you happen to know why this doc has a In this article....

This can be disable in [metadata.json](https://github.com/MonoGame/MonoGame/blob/develop/Documentation/metadata.json) with wildcards. I disabled it for the articles because they're all very short, but forgot the index page.

Jjagg commented on Documentation/articles/introduction/platforms.md:
```diff
@@ -59,20 +74,46 @@ This platform is meant to publish games on the Windows Store, for both Windows a
 The iOS platform uses [Xamarin.iOS](https://docs.microsoft.com/en-us/xamarin/ios/).
 OpenGL is used for graphics, and OpenAL for audio.
+
+### NetStandard Library
```
I don't think these should go here, since they're not platforms. But it's definitely good to have them documented somewhere.

SimonDarksideJ replied:
I've added a new header to denote these are additional project types.  As you say, there isn't really anywhere else they could go.

Jjagg commented on Documentation/articles/packaging_games.md:
```diff
@@ -1,10 +1,14 @@
-### Building and packaging for Windows
+---
+
+## Building and packaging for Windows
```
I think these were supposed to go under the Desktop games section.

SimonDarksideJ replied:
Updated

tomspilman commented:
A few more suggestions...  missing capitalizations, some redundant "MonoGame"s,etc...


![image](https://user-images.githubusercontent.com/103522/88610192-a5bd6980-d04b-11ea-8bb5-2fb9823df777.png)

Jjagg commented:
The FAQ page should be deleted or we should add questions to it. The only question on there right now is already specified in the Requirements section in more detail.

https://docs.monogame.net/articles/introduction/monogame_faq.html

![image](https://user-images.githubusercontent.com/14875382/88631186-0da69b00-d0b2-11ea-8b17-082f96f61d6d.png)

Jjagg commented:
Couple of broken links here http://teamcity.monogame.net/viewLog.html?buildTypeId=MonoGame_GenerateDocumentation&buildId=99979&tab=buildLog&logTab=tree&filter=debug&expand=all&_focus=104.

SimonDarksideJ commented:
Fixed those broken links @Jjagg , oddly they were in code and not the documentation, but cleaned up anyway.

SimonDarksideJ commented:
Latest feedback addressed

Jjagg commented:
> Fixed those broken links @Jjagg , oddly they were in code and not the documentation, but cleaned up anyway.

Yeah, it wasn't anything that was broken in this PR, I just noticed the warnings in the build log.

SimonDarksideJ commented:
Anything else needed @tomspilman / @Jjagg to bring this to a conclusion?

Jjagg commented:
I haven't looked at this with a browser yet! Will get to that tomorrow

SimonDarksideJ commented:
Link here @Jjagg  / @tomspilman 
http://teamcity.monogame.net/repository/download/MonoGame_GenerateDocumentation/100012:id/Documentation.zip!/index.html

Jjagg commented:
With regards to the getting_started/creating_a_new_project_* pages:

We shouldn't duplicate requirements and tooling installation information on here. We also shouldn't put stuff like "before 3.8 we used an installer". Stuff like that gets dated and irrelevant and I specifically wrote the migration page to provide that installation.

I actually prefer the create_project page that was deleted in this PR. It's a lot more concise and was written specifically for 3.8 already. It could do with some images though.

I've gone through all docs and edited a bunch of stuff, will set up a PR to your branch shortly.

tomspilman commented:
I'm good with the changes... so @Jjagg call on all this.

We can go on forever improving these, so we need to decide when this PR is good enough and cut it off.

Jjagg commented:
I just set up a big PR with lots of small fixes/improvements: SimonDarksideJ/MonoGame#8

SimonDarksideJ commented:
Thanks for the updates @Jjagg much appreciated.

Jjagg commented:
Can I get some input on the creating_a_new_project_* pages? (Check my previous comment)

Other than that this is good to go IMO.

SimonDarksideJ commented:
How does this sound @Jjagg , I've removed the old mgcb-editor instructions from each getting started guide and instead added the following reference, as the instructions (as you point out) are included in the guide for the editor:

```
## MonoGame Content Pipeline Tool (MGCB Editor)

The MGCB Editor is also published to the .NET tools library. To install the MGCB Editor please [follow the instructions here](~/articles/tools/mgcb_editor.md#installation_instructions).

For more details on the [MonoGame tools check here](~/articles/tools/tools.md).

```

If you're ok with that, I'll commit and publish it.

I still believe you need separate guides for the three development types as each is unique in its setup and screens.  The follow on guides are the same however.  But I agree, we should avoid duplication of instructions for the sake of duplication, where it's is specifically designed,.

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
 # Creating a Project with Visual Studio
 
-Start up Visual Studio and select **New Project...** in the upper left corner.
+MonoGame is available on Windows for development using the instructions below to get started with MonoGame 3.8.
```
This line is redundant.

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
-Start up Visual Studio and select **New Project...** in the upper left corner.
+MonoGame is available on Windows for development using the instructions below to get started with MonoGame 3.8.
+
+> Note, MonoGame 3.8 projects are no longer compatible with earlier versions of MonoGame.  If you wish to work on or build older MonoGame projects, then you will still need to install [MonoGame 3.7.1](https://www.monogame.net/downloads/) or earlier to open them.
```
This isn't true, MG < 3.8 is compatible with 3.8+. It's just the setup process that's changed.

SimonDarksideJ replied:
Not true.  Without the Monogame 3.7.1 MSI install, this is what I get when I open 3.7 solutions with only the Monogame 3.8 templates installed
![image](https://user-images.githubusercontent.com/1793042/89084882-f3f2a580-d388-11ea-9726-5f53d748ba1d.png)

Jjagg replied:
Ah, I misunderstood what you meant by incompatible. Yes, old projects require the SDK install. But I wouldn't call old projects incompatible with 3.8, because upgrading is easy (and I'd say recommended to get all .NET Core goodies).

But... this part of the guide is for setting up a new project, which I would very, very strongly recommend everyone does with 3.8 in the future. I don't think we need to bring up the older versions or how things changed here.

I would move this information to the migration guide. Just a short note, something like:

> If you do not want to upgrade your project you'll need to keep the MonoGame SDK installed.

When we announce the release we should be clear that people need to check out the migration page if they have old projects and want to upgrade to 3.8.

SimonDarksideJ replied:
Ok, that's fair enough. But we need to ensure the message is clear.  (and that include our own samples until I get them updated, soon (tm) )

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
+
+![Visual Studio optional components](~/images/getting_started/1_installer_vs_components.png)
+
+* .Net Desktop Development - For Windows GL / DX platforms
```
This isn't necessary, unless you want to target .NET Framework with WindowsDX to support old Windows versions or to ship without an embedded .NET runtime. The .NET Core payload is required.

SimonDarksideJ replied:
Maybe I'll reword, but without this component you also would never be able to open older 3.7 packages.  But I grant you for the netcore/netstandard projects it may not be required, I'll have to test.

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
+
+![Visual Studio optional components](~/images/getting_started/1_installer_vs_components.png)
+
+* .Net Desktop Development - For Windows GL / DX platforms
+* Mobile Development with .NET - For Android / iOS platforms
+* Universal Windows Platform development - For Windows 10 / Xbox UWP
+
+When installing Visual Studio, it also is recommended to include the "**.NET Core cross-platform development" components:
+
+![DotNet Core component](~/images/getting_started/1_netcorecomponet.png)
+
+> Alternatively, you can specifically install the [.NET Core SDK from here](https://dotnet.microsoft.com/download), SDK Versions 3.1 and above.
```
All of this information is on the [requirements page](https://github.com/MonoGame/MonoGame/blob/5ddc1518ebf4eacecc5e454382967363c209e6a5/Documentation/articles/introduction/requirements.md).

SimonDarksideJ replied:
Not all, but I get your point. I'll ensure all the pre-requisites for all platforms are moved to that page.
But I'll also have to create a MAC section as well.

If you are ok with that?

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
+
+## Install MonoGame Templates
+
+From MonoGame 3.8, we no longer use MSI Installers for deploying the framework and templates as this creates hard dependencies to specific machine setups or DLL locations.  Instead, MonoGame now uses NuGet and the DotNet deployment platform to reference a specific version of MonoGame online.
```
This information is on the migration page. IMO we shouldn't mention the old setup in any other page. It would be better to link the migration page if we think that's necessary.

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
 ![New Solution](~/images/getting_started/1_new_soulution_vs.png)
 
-Now you should see a "New Project" dialog pop up, from here select **Templates > Visual C# > MonoGame** category, and then select **MonoGame Cross Platform Desktop Project**. Next type in the name that you wish to give your project, for this tutorial let's just use **ExampleGame** (do note that you should not use space character for it). After you've entered the name, click on the **Browse** butt ...
+Now you should see a "New Project" dialog pop up, from here select **Templates > Visual C# > MonoGame** category, and then select **MonoGame Cross Platform Desktop Project**. Next type in the name that you wish to give your project, for this tutorial let's just use **ExampleGame** (*note* you should not use spaces when naming the project). After you've entered the name, click on the **Browse** bu ...
```
This is for VS2017. The new project wizard got an overhaul for VS2019.

SimonDarksideJ replied:
I'll double-check as I did update a lot of the assets for 2019/3.8

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
+
+When installing Visual Studio, it also is recommended to include the "**.NET Core cross-platform development" components:
+
+![DotNet Core component](~/images/getting_started/1_netcorecomponet.png)
```
.NET Core

Jjagg commented on Documentation/articles/getting_started/1_creating_a_new_project_vs.md:
```diff
@@ -1,19 +1,75 @@
+
+## Prerequisites
+
+You will need a copy of [Visual Studio 2019](https://www.monogame.net/downloads/) or later installed (any edition, including Community) before installing MonoGame, with the following components (depending on your target platform):
```
I would get rid of the 'or later' part. We'll need to make sure things work with the next version of Visual Studio and we should update our docs when we do that.

SimonDarksideJ replied:
Prereq's moving to the development req's page instead.

Jjagg commented:
> The MGCB Editor is also published to the .NET tools library.

- At this point in the guide users don't know what MGCB Editor is or why they need it.
- There's no .NET tools library, .NET Core tools are published to NuGet.
- The latest version is **only** published as a .NET Core tool.

> My only other concern in this area is we have one guide to describe and install the mgcb-editor:

I think this is fine. One's the installation guide and the other is in-depth information on how the tool works and can be used.

SimonDarksideJ commented:
> At this point in the guide users don't know what MGCB Editor is or why they need it.

Fair point @Jjagg , I'll remove it from that point in the instructions and I've added the following note to the "Adding Content" page

```
```

SimonDarksideJ commented:
Addressed all feedback @Jjagg with the exception of the Monogame 3.7 content.  As tested, the MonoGame 3.8 templates installed in Visual Studio for 3.8 are incompatible with 3.7. As such, 3.7 projects cannot be opened without the previous MSI installer loaded.

SimonDarksideJ commented:
> I also think the project creation guide shouldn't be a part of the getting started guide

I'd disagree here @Jjagg as the only users likely to need these instructions in reality, are those new to MonoGame and just "Getting Started" :D

SimonDarksideJ commented:
Any other comments @Jjagg  to close this one out?

Jjagg commented:
> I'd disagree here @Jjagg as the only users likely to need these instructions in reality, are those new to MonoGame and just "Getting Started" :D

I imagine people that are familiar with MonoGame only need to check out how to create a new project with 3.8+.

> Any other comments @Jjagg to close this one out?

There’s probably always something to improve, but I don’t want to block this any longer, and honestly these changes are great!
Thanks Simon! 🙏😊
